bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/dao/impl/PatientDaoImpl.java

```diff
@@ -77,24 +81,35 @@ public PatientDaoImpl(SessionFactory sessionFactory) {
                                                               Boolean filterPatientsByLocation, Boolean filterOnAllIdentifiers) {
-
         validateSearchParams(customAttributeFields, programAttributeFieldName, addressFieldName);
 
-        List<PatientIdentifier> patientIdentifiers = getPatientIdentifiers(identifier, filterOnAllIdentifiers, offset, length);
-        List<Integer> patientIds = patientIdentifiers.stream().map(patientIdentifier -> patientIdentifier.getPatient().getPatientId()).collect(toList());
-        Map<Object, Object> programAttributes = Context.getService(BahmniProgramWorkflowService.class).getPatientProgramAttributeByAttributeName(patientIds, programAttributeFieldName);
+        Set<Patient> patients = new HashSet<>();
+
+        if(identifier != null && !identifier.isEmpty()){
+            patients.addAll(patientService.getPatients(identifier, false, offset, length));
+            if(name != null && !name.isEmpty()){
+                patients.retainAll( patientService.getPatients(name, false, offset, length));
+            }
+            if(customAttribute != null && !customAttribute.isEmpty()){
+                patients.retainAll( patientService.getPatients(customAttribute, false, offset, length));
+            }
+        }else if(name != null && !name.isEmpty()){
+            patients.addAll(patientService.getPatients(name, false, offset, length));
+            if(customAttribute!=null && !customAttribute.isEmpty()){
+                patients.retainAll(patientService.getPatients(customAttribute, false, offset, length));
+            }
+        }else if((customAttribute != null && !customAttribute.isEmpty())){
+            patients.addAll(patientService.getPatients(customAttribute, false, offset, length));
+        }
+
+        List<Integer> patientIds = patients.stream().map(Patient::getPatientId).collect(toList());
         PatientResponseMapper patientResponseMapper = new PatientResponseMapper(Context.getVisitService(),new BahmniVisitLocationServiceImpl(Context.getLocationService()));
-        Set<Integer> uniquePatientIds = new HashSet<>();
-        List<PatientResponse> patientResponses = patientIdentifiers.stream()
-                .map(patientIdentifier -> {
-                    Patient patient = patientIdentifier.getPatient();
-                    if(!uniquePatientIds.contains(patient.getPatientId())) {
-                        PatientResponse patientResponse = patientResponseMapper.map(patient, loginLocationUuid, patientSearchResultFields, addressSearchResultFields,
-                                programAttributes.get(patient.getPatientId()));
-                        uniquePatientIds.add(patient.getPatientId());
-                        return patientResponse;
-                    }else
-                        return null;
+        Map<Object, Object> programAttributes = Context.getService(BahmniProgramWorkflowService.class).getPatientProgramAttributeByAttributeName(patientIds, programAttributeFieldName);
```

Comment:
should autowire this service in the constructor instead of doing Context.getService